client.js

```diff
@@ -0,0 +1,427 @@
+        }
+        break;
+      case Container.Type.eCriterionLimitsResponse:
+        data.criterionLimitsResponse.criterionMin += this.timeDiff;
+        data.criterionLimitsResponse.criterionMax += this.timeDiff;
+        const limits = {
+          start_s: data.criterionLimitsResponse.criterionMin,
+          end_s: data.criterionLimitsResponse.criterionMax
+        };
+        if (this.storedPromises[data.criterionLimitsResponse.requestId]) {
+          const { resolve } = this.storedPromises[data.criterionLimitsResponse.requestId];
+          delete this.storedPromises[data.criterionLimitsResponse.requestId];
+          resolve(limits);
+        }
+        break;
+      case Container.Type.eVersionResponse:
+        const version = parseFloat(data.versionResponse.version);
+        if (version < 3.0) {
+          if (this.storedPromises[data.versionResponse.requestId]) {
+            const { reject } = this.storedPromises[data.versionResponse.requestId];
+            delete this.storedPromises[data.versionResponse.requestId];
+            reject(new Error("CDP version needs to be 3.0 or newer."));
+          }
+        } else {
+          if (this.storedPromises[data.versionResponse.requestId]) {
+            const { resolve } = this.storedPromises[data.versionResponse.requestId];
+            delete this.storedPromises[data.versionResponse.requestId];
+            resolve(data.versionResponse.version);
+          }
+        }
+        break;
+      case Container.Type.eSignalDataResponse:
+        let dataPoints = [];
+        let index = 0;
+        for (const row of data.signalDataResponse.row) {
+          data.signalDataResponse.criterion[index] += this.timeDiff;
+          let signalNames = [];
+          for (const signalId of row.signalId) {
+            signalNames.push(this.idToName[signalId]);
+          }
+          const value = this._createValue(
+            signalNames,
+            row.minValues,
+            row.maxValues,
+            row.lastValues
+          );
+          dataPoints.push({
+            timestamp: data.signalDataResponse.criterion[index],
+            value: value
+          });
+          index++;
+        }
+        if (this.storedPromises[data.signalDataResponse.requestId]) {
+          const { resolve } = this.storedPromises[data.signalDataResponse.requestId];
+          delete this.storedPromises[data.signalDataResponse.requestId];
+          resolve(dataPoints);
+        }
+        break;
+      default:
+        console.error("Unknown message type", data.messageType);
+    }
+  }
+
+  _createValue(signalNames, minValues, maxValues, lastValues) {
+    const value = {};
+    for (let i = 0; i < signalNames.length; i++) {
+      value[signalNames[i]] = {
+        min: this._valueFromVariant(minValues[i]),
+        max: this._valueFromVariant(maxValues[i]),
+        last: this._valueFromVariant(lastValues[i])
+      };
+    }
+    return value;
+  }
+
+  _valueFromVariant(value) {
```

Comment:
Noticed the real-time client uses a different method for valueFromVariant conversions.

https://github.com/CDPTechnologies/JavascriptCDPClient/blob/main/index.js#L94